src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/DebugFrameworksDynamicMenuCommand.cs

```diff
@@ -106,7 +106,9 @@ public override bool QueryStatusCommand(int cmdIndex, EventArgs e)
         /// </summary>
         protected virtual void ExecuteSynchronously(Func<Task> asyncFunction)
```

Comment:
❓ Why can this not be changed to this?

```csharp
protected virtual Task ExecuteAsync(Func<Task> asyncFunction)
{
  return await asyncFunction().ConfigureAwait(false);
}
```

This change would allow tests to override the implementation and also correct the underlying warning without a suppression. The use of `JoinableTaskFactory.Run` in the caller of this method should not be an impediment to testing, especially if test code overrides this method to return the result synchronously.

Reply:
ThreadHelper expects to be initialized on a pumping UI thread, this avoids that.

Reply:
:bulb: This would be good to put in the documentation for the method.